presto-main/src/main/java/com/facebook/presto/operator/SortedPositionLinks.java

```diff
@@ -165,7 +168,7 @@ private SortedPositionLinks(PositionLinks positionLinks, int[][] sortedPositionL
         this.positionLinks = requireNonNull(positionLinks, "positionLinks is null");
         this.sortedPositionLinks = requireNonNull(sortedPositionLinks, "sortedPositionLinks is null");
         this.lessThanFunction = requireNonNull(lessThanFunction, "lessThanFunction is null");
-        this.sizeInBytes = positionLinks.getSizeInBytes() + sizeOf(sortedPositionLinks);
+        this.sizeInBytes = INSTANCE_SIZE + positionLinks.getSizeInBytes() + (SizeOf.sizeOf(sortedPositionLinks[0]) * sortedPositionLinks.length);
```

Comment:
I don't think so. I wanted to get a better approximation than what we have right now. If we want to get this totally accurate we need to traverse the array.